modules/mono/glue/GodotSharp/GodotSharp/Core/Object.base.cs

```diff
@@ -132,6 +132,22 @@ public SignalAwaiter ToSignal(Object source, StringName signal)
             return new SignalAwaiter(source, signal, this);
         }
 
+        /// <summary>
+        /// Calls the specified <see cref="Action"/> during idle time. Example:
+        /// <code>
+        /// public override void _Process(float delta)
+        /// {
+        ///     var collisionShape = GetNode&lt;CollisionShape2D&gt;("CollisionShape2D");
+        ///     Defer(() => collisionShape.Disabled = true);
+        /// }
+        /// </code>
+        /// </summary>
+        /// <param name="callback">Callback to call during idle time.</param>
+        protected void Defer(Action callback)
```

Comment:
I prefer to name this `CallDeferred` (to make it an overload), unless GDScript has a similar function/keyword called `defer`.

Reply:
The reason I changed it from `CallDeferred` was to make the distinction between the `public CallDeferred` and the protected lambda function. Happy to change it to an overload, should probably also make it public at that point. Opens up the opportunity of people doing things like `_myNode.CallDeferred(() => _myOtherNode.Disabled = true);`. Which gives me the fear 😆

Reply:
Shouldn't the method be static?

Reply:
@neikeq You're right; it probably should. Good catch.

So just to summarize changes I need to make:
1. Change `Defer` to `CallDeferred`?
2. Make method static.
3. Do we keep it `protected`, or is that too confusing when there are `public` overloads?
4. Do we want/need to support inferred returns as somebody mentioned? eg. `var isDisabled = CallDeferred(() => _colisionShape.IsDisabled)`? (I'm not convinced we do)